platforms/documentation/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy

```diff
@@ -320,7 +321,10 @@ class SamplesJavaTestingIntegrationTest extends AbstractSampleIntegrationTest {
         def result = succeeds("test", "integrationTest")
 
         then:
-        result.assertTaskOrder(":test", ":integrationTest")
+        // Task order is non-deterministic in CC
```

Comment:
❓ Isn't there [shouldRunAfter](https://github.com/gradle/gradle/blob/0077a27b809594555944ee18b3dc334d9aafa5d5/platforms/documentation/docs/src/snippets/java/basic/groovy/build.gradle#L115) in the build file? However, it looks like should successors are mostly ignored for CC and the task still may start if all its hard dependencies completed. Maybe we should document this, for example as "Gradle will prefer to start `shouldRunAfter` argument before this task but will not wait for it to complete before starting this task if parallel execution is allowed"?

Reply:
It is: https://github.com/gradle/gradle/blob/c42798b4690cbec3294ff3d927c296e88749fa3b/platforms/documentation/docs/src/snippets/java/basic/groovy/build.gradle#L115 

Will open a documentation issue.

Reply:
https://github.com/gradle/gradle/issues/29862